src/CurrentRoute.php

```diff
@@ -77,4 +82,23 @@ public function setUri(UriInterface $uri): void
         }
         throw new RuntimeException('Can not set URI since it was already set.');
     }
+
+    public function getParameters(): array
+    {
+        return $this->parameters;
+    }
+
+    public function getParameter(string $name, $default = null): ?string
+    {
+        return $this->parameters[$name] ?? $default;
+    }
+
+    public function setParameters(array $parameters)
+    {
+        if ($this->parameters === []) {
+            $this->parameters = $parameters;
+            return;
+        }
+        throw new RuntimeException('Can not set parameters since it was already set.');
```

Comment:
May be throw `LogicException`?

From docs:

`LogicException` that represents error in the program logic. This kind of exception should lead directly to a fix in your code.

`RuntimeException` thrown if an error which can only be found on runtime occurs.